### ISSUE TITLE
Cuthbert Gaussian filters transform their Cholesky factor to a dense covariance unnecessarily

### ISSUE THREAD

**Issue by MatthieuDarcy** [BasisResearch/dynestyx#357]
Low priority change, small optimization gain. 

## Summary

For cuthbert-backed Gaussian filters (`KFConfig`, `EKFConfig`, and full-rank `EnKFConfig`), `_cholesky_state_sequence_to_dists` squares the Cholesky factor it already has, then hands the product to `MultivariateNormal`, which factors it back. That is an avoidable `O(T * d^3)` cost + precision loss.

The underlying reason is because`_gaussian_sequence_to_dists` is shared by Cuthbert (Cholesky factor) and CD-Dynamax (covariance). Cuthbert filters transform their cholesky factors to match the CD-Dynamax convention.

## Where

`dynestyx/inference/utils/distribution_utils.py:300`:

```python
covariances = covariance_from_cholesky(states.chol_cov)   # chol @ chol.T
...
return _gaussian_sequence_to_dists(states.mean, covariances, plate_shapes=plate_shapes)
```

and `_gaussian_sequence_to_dists` (`:118`) then does:

```python
dist.MultivariateNormal(
    _slice_time_axis(means, t, plate_shapes),
    covariance_matrix=_slice_time_axis(covariances, t, plate_shapes),
)
```

numpyro's `MultivariateNormal.__init__` computes `scale_tril = jnp.linalg.cholesky(covariance_matrix)` eagerly.

## Suggested fix

Let the helper accept either parameterisation and pass through whichever the caller actually holds:

```python
def _gaussian_sequence_to_dists(
    means, covariances=None, *, scale_trils=None, plate_shapes=(), ...
):
```

cuthbert passes `scale_trils=states.chol_cov`; cd-dynamax passes `covariances=` unchanged. 

**Wrinkle:** on #356, a full-rank EnKF uses a nonzero jitter by default. `recorded_filtered_states_cov_jitter` is applied as
`covariances + eps * I`, which has no direct `scale_tril` equivalent —
`chol(L @ L.T + eps * I) != L + something`. Options:

- fall back to the dense path when the jitter is nonzero (leaves the
  full-rank EnKF on the slow path by default).
- set the jitter to be 0.0 when EnKF is full rank. Restricting the jitter to the low-rank branch
  would make the fast path unconditional here, at the cost of the field no longer
  meaning the same thing in both branches.
